src/adapter/src/flags.rs

```diff
@@ -43,6 +43,7 @@ pub fn storage_config(config: &SystemVars) -> StorageParameters {
             tcp_user_timeout: Some(config.pg_source_tcp_user_timeout()),
         },
         pg_source_snapshot_statement_timeout: config.pg_source_snapshot_statement_timeout(),
+        pg_source_wal_sender_timeout: config.pg_source_wal_sender_timeout(),
```

Comment:
setting flags this way has been deprecated -- would switch this to use `dyncfg` instead -- also a lot less boilerplate!

Reply:
If this has been deprecated, why are all of these options still set this way? I also didn't see an announcement that these code paths have been deprecated––can you link me to it?

Reply:
https://materializeinc.slack.com/archives/C01CFKM1QRF/p1710780368452619 and https://materializeinc.slack.com/archives/CUXUBJH7B/p1710192018978979

We want to move these over at some point -- we just haven't done so yet, but we shouldn't keep adding new ones if we can avoid it

Reply:
I guess given that this is setting a timeout closely related to the other pg source timeout settings it's fine to leave as-is and we should migrate them all over at some point

Reply:
Nice––I hadn't seen Gus's announcement. None of those spots mention "deprecation," though. Aslo, this PR is green and the work to move this option alongside all of the others is nominal, so don't see the value in re-running CI for this. Next time I need to add an option, though, will plumb it through `dyncfg`.

Reply:
> I guess given that this is setting a timeout closely related to the other pg source timeout settings it's fine to leave as-is and we should migrate them all over at some point

agreed!